## Create MapReduce Funcation

In [3]:
%%file Categry_Sector_Count.py
 # import the mrjob library
from mrjob.job import MRJob

# creat the class
class ChangeCategry_Sector_Count(MRJob): 
    
    
    # in the map step each row in dataframe is read key for each value pair
    def mapper(self,_,sector_change ):
     
        yield (sector_change, 1) 
        # The output for each row as a tuple 
        

    # in the reduce step combine all tuples with the same key
    # after that  sum all the values of the tuple, which will give the total 
    def reducer(self, key, values):
        yield (key, sum(values))
        
if __name__ == "__main__":
    ChangeCategry_Sector_Count.run()

Overwriting Categry_Sector_Count.py


## Run the Funcation

In [4]:
!python Categry_Sector_Count.py data.csv

No configs found; falling back on auto-configuration
No configs specified for inline runner
Creating temp directory /var/folders/9l/pggd3pz94nx1xsh1q0qkty3c0000gn/T/Categry_Sector_Count.fatimah.20221129.083518.344991
Running step 1 of 1...
job output is in /var/folders/9l/pggd3pz94nx1xsh1q0qkty3c0000gn/T/Categry_Sector_Count.fatimah.20221129.083518.344991/output
Streaming final output from /var/folders/9l/pggd3pz94nx1xsh1q0qkty3c0000gn/T/Categry_Sector_Count.fatimah.20221129.083518.344991/output...
"Materials,Good Change"	61414
"Materials,Stable"	16277
"Real Estate,Bad Change"	19975
"Energy,Bad Change"	7202
"Energy,Good Change"	6904
"Energy,Stable"	1625
"Financials,Bad Change"	67244
"Financials,Good Change"	63151
"Financials,Stable"	14455
"Health Care,Bad Change"	8259
"Health Care,Good Change"	8018
"Health Care,Stable"	1548
"IT,Bad Change"	394
"IT,Good Change"	360
"IT,Stable"	52
"Industrials,Bad Change"	28957
"Industrials,Good Change"	27568
"Industrials,Stable"	7412
"Materials,Bad Chan

#### We found the highest count is
#### Financials with Bad Change"	67244

#### The lowest count is
#### IT sector wuth stable change =52